src/Core/tests/DeviceTests/Services/FontManagerTests.Android.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Android.App;
+using Android.Graphics;
+using Java.Interop;
+using Xunit;
+
+namespace Microsoft.Maui.DeviceTests;
+
+[Category(TestCategory.Fonts)]
+public partial class FontManagerTests : TestBase
+{
+	[Theory]
+	[InlineData("dokdo_regular", "dokdo_regular.ttf")]
+	[InlineData("dokdo_regular.ttf", "dokdo_regular.ttf")]
+	[InlineData("myalias", "dokdo_regular.ttf")]
+	[InlineData("insubfolder", "Fonts/insubfolder.ttf")]
+	[InlineData("insubfolder.ttf", "Fonts/insubfolder.ttf")]
+	public void CanLoadFonts(string fontName, string assetName)
+	{
+		var fontWeight = FontWeight.Regular;
+		var registrar = new FontRegistrar(fontLoader: null);
+		registrar.Register("dokdo_regular.ttf", "myalias");
+		var manager = new FontManager(registrar);
+		var actual = manager.GetTypeface(Font.OfSize(fontName, 12, fontWeight));
+		var expected = Typeface.Create(Typeface.CreateFromAsset(Application.Context.Assets, assetName), (int)fontWeight, italic: false);
+		Assert.True(expected.Equals(actual));
```

Comment:
The `equals` method checks this `native_instance` field:

https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/graphics/java/android/graphics/Typeface.java#194

So if you have a matching path & font weight, it will return `true`.